daemon.go

```diff
@@ -45,19 +46,15 @@ func (d *Daemon) Start(ctx context.Context) error {
 	if err := d.Queue.Create(); err != nil {
 		return err
 	}
-	defer func() {
-		if err := d.Queue.Delete(); err != nil {
-			log.WithError(err).Error("Failed to delete queue")
-		}
-	}()
 
 	if err := d.Queue.Subscribe(); err != nil {
 		return err
 	}
+
+	// ensure the queue deletion happens only once
+	var deleteOnce sync.Once
 	defer func() {
```

Comment:
Since the deferred call happens after `Queue.Subscribe()`, we could leak the SQS queue if we fail to subscribe (which is likely to happen since the topic ARN is a user argument).